src/include/migraphx/op/slice.hpp

```diff
@@ -66,7 +66,8 @@ struct slice
 
     std::string name() const { return "slice"; }
 
-    // Convert negative index to a value counting backwards from the max index, per numpy convention
+    // Convert negative index to a value counting backwards from the max index, per numpy slicing
+    // convention, and clip large values to 1 more than the largest index.
```

Comment:
This comment is a bit confusing.
It is clipping larger values to the largest possible value. not 1 more. 

also we should check if `index` is within bounds for negative values. 
e.g. for the dimensions `7` if the index is -8.

Reply:
The comment may be a bit confusing, but it's accurate.  The largest possible value is 1 more than the largest index.

Reply:
I understand what fix_index() is trying to do now. The comment could say that the end is exclusive.